compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEAssignNpuDmaBdIds.cpp

```diff
@@ -177,40 +188,63 @@ FailureOr<AMDAIE::BdIdOp> getBdIdOp(
     // `offset` is the BD ID assigned to current DMA op at first iteration.
     Value iv = loop.getInductionVar();
 
-    // Get the number of BD IDs will be assigned to current DMA op.
+    // Step 1: Get the number of BD IDs will be required by the current DMA op.
+    // Also fetch other DMA ops within the same scf.for block that are operating
+    // on the same tile.
     uint32_t numRequired = 0;
-    getNumRequiredBdIds(loop, npuDmaOp, tileOp, shimTileToGeneratorMap,
-                        numRequired);
+    SmallVector<AMDAIE::NpuDmaCpyNdOp> dmaOps = getNumRequiredBdIdsAndDMAOps(
+        loop, npuDmaOp, tileOp, shimTileToGeneratorMap, numRequired);
+
+    // Step 2: Compute BD ID split amongst all DMA ops.
     uint32_t numAvailable = generator.getNumAvailableBdIds(channel);
+    uint32_t totalDmaOps = dmaOps.size();
     uint32_t size = std::max(numAvailable / numRequired, 1u);
-
-    // Only create expression if more than 1 BD ID is needed,
-    // otherwise, fall back to constant BD ID.
-    if (size > 1) {
-      // Assigning BD IDs for all iterations in the loop.
-      SmallVector<uint32_t> bdIds;
-      for (uint32_t i = 0; i < size; i++) {
+    // In case an equal split of BD IDs is not possible, return failure.
```

Comment:
This seems not about an equal split, but in case the there are not enough BD ids available?

Reply:
Done.